src/main/kotlin/no/fdk/dataset/preview/service/PreviewService.kt

```diff
@@ -64,6 +69,45 @@ class PreviewService(
         }
     }
 
+    private fun xlsxPreview(rows: Int?, body: ResponseBody): Preview {
+        logDebug("Parsing Excel")
+
+        val tableRows = arrayListOf<TableRow>()
+
+        val workbook: Workbook = XSSFWorkbook(body.byteStream())
+        val formatter = DataFormatter()
+        val sheet = workbook.getSheetAt(0)
+
+        var lastCellNum = 0
+        sheet.forEach { row ->
+            val tableRow = TableRow(row.map {
+                formatter.formatCellValue(it)
+            })
+
+            tableRows.add(tableRow)
+
+            lastCellNum = when {
+                row.lastCellNum <= lastCellNum -> lastCellNum
+                formatter.formatCellValue(row.last()).isNotEmpty() -> row.physicalNumberOfCells
+                else -> lastCellNum
+            }
+        }
+
+        val headerIndex = tableRows.indexOfFirst {
+            it.columns.size == lastCellNum && it.columns[lastCellNum-1].isNotEmpty()
+        }
+        val start = if (headerIndex == -1) 0 else headerIndex
+        val end =
```

Comment:
Kanskje kalkulere `start+1+getMaxNumberOfRows(rows)` og `tableRows.size-1` på forhånd?
```suggestion
        val maxRows = start + 1 + getMaxNumberOfRows(rows)
        val tableSize = tableRows.size - 1
        val end = if (maxRows <= tableSize) maxRows else tableSize
```